hoomd/VectorVariant.h

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) 2009-2023 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+// Copyright (c) 2009-2023 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+#pragma once
+
+#include <cstdint>
+#include <pybind11/pybind11.h>
+#include <utility>
+
+#include "BoxDim.h"
+#include "HOOMDMath.h"
+#include "Variant.h"
+
+namespace hoomd
+    {
+/** Defines vector quantities that vary with time steps.
+
+    VectorVariant provides an interface to define vector quanties (such as box dimensions) that vary
+   over time. The base class provides a callable interface. Derived classes implement specific kinds
+   of varying quantities.
+*/
+template<unsigned int ndim> class PYBIND11_EXPORT VectorVariant
+    {
+    public:
+    virtual ~VectorVariant() { }
+    typedef std::array<Scalar, ndim> array_type;
+
+    /** Return the value of the Variant at the given time step.
+
+        @param timestep Time step to query.
+        @returns The value of the variant.
+    */
+    virtual array_type operator()(uint64_t timestep)
+        {
+        std::array<Scalar, ndim> ret;
+        ret.fill(0);
+        return ret;
+        }
+    };
+
+class PYBIND11_EXPORT VectorVariantBox : public VectorVariant<6>
+    {
+    protected:
+    static std::array<Scalar, 6> box_to_array(std::shared_ptr<BoxDim> box)
+        {
+        return std::array<Scalar, 6> {box->getL().x,
+                                      box->getL().y,
+                                      box->getL().z,
+                                      box->getTiltFactorXY(),
+                                      box->getTiltFactorXZ(),
+                                      box->getTiltFactorYZ()};
+        }
+    };
+
+class PYBIND11_EXPORT VectorVariantBoxConstant : public VectorVariantBox
+    {
+    public:
+    /** Construct a VectorVariantBoxConstant.
+
+        @param box The box.
+    */
+    VectorVariantBoxConstant(std::shared_ptr<BoxDim> box) : m_box(box) { }
+
+    virtual ~VectorVariantBoxConstant() { }
+
+    /// Return the value.
+    virtual array_type operator()(uint64_t timestep)
+        {
+        return box_to_array(m_box);
+        }
+
+    std::shared_ptr<BoxDim> getBox()
+        {
+        return m_box;
+        }
+
+    void setBox(std::shared_ptr<BoxDim> box)
+        {
+        m_box = box;
+        }
+
+    protected:
+    std::shared_ptr<BoxDim> m_box;
+    };
+
+class PYBIND11_EXPORT VectorVariantBoxInterpolate : public VectorVariantBox
+    {
+    public:
+    /** Construct a VectorVariantBoxInterpolate to interpolate between two boxes linearly in time.
+
+        @param box1 The initial box
+        @param box2 The final box
```

Comment:
The python code calls these `initial_box` and `final_box`. Please rename all instances of `box1` to `initial_box` and all instances of `box2` to `final_box` for consistency.